File: ui/src/models/ApprovalRequestTask.ts
import { IApprovalRequest } from "./approvalRequest";
import { ApprovalStatus } from "./approvalStatus";

export interface IApprovalRequestTask {
  id: number;
  approvalRequest: IApprovalRequest;
  approver: string;
  status: ApprovalStatus;
  completed?: string;
  completedDate?: Date;
  comment?: string;
}

File: api/src/Models/ApprovalRequestTask.cs
namespace api.Models;

public class ApprovalRequestTask
{
    public long Id { get; set; }
    public required ApprovalRequest ApprovalRequest { get; set; }
    public required string Approver { get; set; }
    public ApprovalStatus Status { get; set; }
    public string? Comment { get; set; }
}
